### PR TITLE
[doc] Fix doxygen off-by-one comments, fix typos

### PR DESCRIPTION
**Pull request by maflcko**
Also
- Make import zmq error fatal again (instead of silent ignore)
- Make `fix-copyright-headers.py` more portable

### CODE REVIEW

**Comment by MarcoFalke** [comment]
Example of comments being of by one: https://dev.visucore.com/bitcoin/doxygen/chain_8h.html#a0d8c285f70a59a32f4f6ab41b78f93ada722a05d892a13ada5ffd4fcbbef837cd

**Comment by luke-jr** [comment]
ACK

File: src/script/sign.h [hunk 1]
```diff
@@ -51,7 +51,7 @@ class MutableTransactionSignatureCreator : public TransactionSignatureCreator {
     MutableTransactionSignatureCreator(const CKeyStore* keystoreIn, const CMutableTransaction* txToIn, unsigned int nInIn, const CAmount& amount, int nHashTypeIn) : TransactionSignatureCreator(keystoreIn, &tx, nInIn, amount, nHashTypeIn), tx(*txToIn) {}
 };
 
-/** A signature creator that just produces 72-byte empty signatyres. */
```
Comment (laanwj) [inline src/script/sign.h]: SignaTyres™

File: contrib/devtools/fix-copyright-headers.py [hunk 2]
```diff
@@ -16,7 +16,7 @@
 import re
 
 year = time.gmtime()[0]
-CMD_GIT_DATE = 'git log --format=@%%at -1 %s | date +"%%Y" -u -f -'
+CMD_GIT_DATE = 'git log --format=%%ad --date=short -1 %s | cut -d"-" -f 1'
```
Comment (laanwj) [inline contrib/devtools/fix-copyright-headers.py]: Whyy oh whyy are we using shell commands in a python script?

**Comment by laanwj** [comment]
utACK

File: contrib/devtools/fix-copyright-headers.py (hunk 2 above)
Comment (MarcoFalke) [inline contrib/devtools/fix-copyright-headers.py]: Lazy<sup>TM</sup> and ItWorks<sup>TM</sup>, also you need to call at least `git` to make this script work. Though, I agree that calling perl is not required. (I won't change it in this pull, though.)

Comment (btcdrak) [inline contrib/devtools/fix-copyright-headers.py]: Make it go full circle. Perl should call Python.

**Comment by btcdrak** [comment]
utACK

File: contrib/devtools/fix-copyright-headers.py (hunk 2 above)
Comment (laanwj) [inline contrib/devtools/fix-copyright-headers.py]: > LazyTM and ItWorksTM, also you need to call at least git to make this script work. 

You don't need to call it through the shell though. Popen, by default, just invokes the process directly. See e.g. how `github-merge.py` interacts with git.

**Pull request merged**